src/QSymbolicsBase/QSymbolicsBase.jl

```diff
@@ -137,25 +138,14 @@ const QObj = Union{AbstractBra,AbstractKet,AbstractOperator,AbstractSuperOperato
 const SymQObj = Symbolic{<:QObj} # TODO Should we use Sym or Symbolic... Sym has a lot of predefined goodies, including metadata support
 Base.:(-)(x::SymQObj) = (-1)*x
 Base.:(-)(x::SymQObj,y::SymQObj) = x + (-y)
+Base.hash(x::SymQObj, h::UInt) = isexpr(x) ? hash(arguments(x), h) : hash([typeof(x),basis(x)], h)
```

Comment:
```suggestion
Base.hash(x::SymQObj, h::UInt) = isexpr(x) ? hash((head(x),arguments(x)), h) : hash((typeof(x),basis(x)), h)
```

The change to the first branch is to make the hash of `X1+X2` and `tensor(X1, X2)` different.

The change to the second branch is less interesting: in julia tuples are much easier for the compiler to optimize away (and not allocate) than vectors. The reason is that tuples are immutable, while vectors potentially can even change size, so tuples can go on the stack, but vectors have to go on the heap. A smarter compiler can in-principle fix that (There are some interesting experimental PR for this at the julia repo, in case you are interested. More generally, modifying the compiler from inside a user session is very easy in julia, so it is a good language if you want to study compiler optimization techniques).

By the way, could we add something more to the second branch so that `SKet(:x)` and `SKet(:y)` also have different tags? Maybe `symbollabel`?